BHoM_UI/Caller/Run.cs

```diff
@@ -0,0 +1,298 @@
+                {
+                    Engine.UI.Compute.RecordExecutionError(e);
+                    return false;
+                }
+            }
+                
+            // Set the output
+            return PushOutputs(result);
+        }
+
+        /*************************************/
+
+        public virtual object Run(List<object> inputs)
+        {
+            if (m_CompiledFunc != null)
+            {
+                try
+                {
+                    return m_CompiledFunc(inputs.ToArray());
+                }
+                catch (InvalidCastException e)
+                {
+                    MethodInfo originalMethod = m_OriginalItem as MethodInfo;
+                    if (originalMethod != null && originalMethod.IsGenericMethod)
+                    {
+                        // Try to update the generic method to fit the input types
+                        MethodInfo method = Engine.Reflection.Compute.MakeGenericFromInputs(originalMethod, inputs.Select(x => x.GetType()).ToList());
+                        m_CompiledFunc = method.ToFunc();
+                        return m_CompiledFunc(inputs.ToArray());
+                    }
+                    else
+                        throw e;
+                }
+            }
+            else if (InputParams.Count <= 0)
+            {
+                BH.Engine.Reflection.Compute.RecordWarning("This is a magic component. Right click on it and <Select a method>");
+                return null;
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The component is not linked to a method.");
+                return null;
+            }
+        }
+
+
+        /*************************************/
+        /**** Helper Methods              ****/
+        /*************************************/
+
+        protected virtual List<object> CollectInputs()
+        {
+            List<object> inputs = new List<object>();
+            try
+            {
+                int index = 0;
+                for (int i = 0; i < m_CompiledGetters.Count; i++)
+                {
+                    object input = null;
+                    try
+                    {
+                        if (InputParams[i].IsSelected)
+                        {
+                            input = m_CompiledGetters[i](m_DataAccessor, index);
+                            index++;
+                        } 
+                        else
+                            input = InputParams[i].DefaultValue;
+                    }
+                    catch (Exception e)
+                    {
+                        Type originalInputType = InputType(m_OriginalItem, i);
+                        if (originalInputType != null && originalInputType.IsGenericType)
+                        {
+                            UpdateInputGenericType(i);
+                            input = m_CompiledGetters[i](m_DataAccessor, index);
```

Comment:
THis this will resolve the issue

```suggestion
                            input = m_CompiledGetters[i](m_DataAccessor, index);
                            index++;
```